src/overlays/actors/ovl_En_Sob1/z_en_sob1.c

```diff
@@ -21,164 +88,1633 @@ const ActorInit En_Sob1_InitVars = {
+        this->stickAnimState = 0;
+    }
+
+    new_var3 = 155.0f * arrowAnimTween;
+
+    this->stickAnimTween = stickAnimTween;
+
+    this->stickLeftPrompt.arrowColorR = COL_CHAN_MIX(255, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorG = COL_CHAN_MIX(new_var2, 155.0f, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorB = COL_CHAN_MIX(0, -100, arrowAnimTween);
+    this->stickLeftPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowColorR = (new_var2 - ((s32)new_var3)) & 0xFF;
+    this->stickRightPrompt.arrowColorG = (255 - ((s32)new_var3)) & 0xFF;
+    this->stickRightPrompt.arrowColorB = COL_CHAN_MIX(0, -100.0f, arrowAnimTween);
+    this->stickRightPrompt.arrowColorA = COL_CHAN_MIX(200, 50.0f, arrowAnimTween);
+
+    this->stickRightPrompt.arrowTexX = 290.0f;
+    this->stickLeftPrompt.arrowTexX = 33.0f;
+
+    this->stickRightPrompt.stickTexX = 274.0f;
+    this->stickRightPrompt.stickTexX += 8.0f * stickAnimTween;
+    this->stickLeftPrompt.stickTexX = 49.0f;
+    this->stickLeftPrompt.stickTexX -= 8.0f * stickAnimTween;
+
+    this->stickLeftPrompt.arrowTexY = this->stickRightPrompt.arrowTexY = 91.0f;
+    this->stickLeftPrompt.stickTexY = this->stickRightPrompt.stickTexY = 95.0f;
+}
+
+s16 EnSob1_GetXZAngleAndDistanceSqToPoint(Path* path, s32 pointIdx, Vec3f* pos, f32* distSq) {
+    Vec3s* points;
+    f32 diffX;
+    f32 diffZ;
+
+    if (path != NULL) {
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        points = &points[pointIdx];
+        diffX = points->x - pos->x;
+        diffZ = points->z - pos->z;
+    } else {
+        diffX = 0.0f;
+        diffZ = 0.0f;
+    }
+    *distSq = SQ(diffX) + SQ(diffZ);
+    return Math_Acot2F(diffZ, diffX) * (0x8000 / M_PI);
+}
+
+void EnSob1_GetCutscenes(EnSob1* this) {
+    this->lookFowardCutscene = this->actor.cutscene;
+    this->lookToShelfCutscene = ActorCutscene_GetAdditionalCutscene(this->lookFowardCutscene);
+    this->lookToShopkeeperCutscene = ActorCutscene_GetAdditionalCutscene(this->lookToShelfCutscene);
+}
+
+void EnSob1_WaitForBlink(EnSob1* this) {
+    s16 decr;
+
+    decr = this->blinkTimer - 1;
+    if (decr != 0) {
+        this->blinkTimer = decr;
+    } else {
+        this->blinkFunc = EnSob1_Blink;
+    }
+}
+
+void EnSob1_Blink(EnSob1* this) {
+    s16 decr = this->blinkTimer - 1;
+    s16 eyeTextureIdxTemp;
+
+    if (decr != 0) {
+        this->blinkTimer = decr;
+        return;
+    }
+    eyeTextureIdxTemp = this->eyeTextureIdx + 1;
+    if (eyeTextureIdxTemp > 2) {
+        this->eyeTextureIdx = 0;
+        this->blinkTimer = (s32)(Rand_ZeroOne() * 60.0f) + 20;
+        this->blinkFunc = EnSob1_WaitForBlink;
+    } else {
+        this->eyeTextureIdx = eyeTextureIdxTemp;
+        this->blinkTimer = 1;
+    }
+}
+
+void EnSob1_Obj3ToSeg6(EnSob1* this, GlobalContext* globalCtx) {
+    gSegments[6] = PHYSICAL_TO_VIRTUAL(globalCtx->objectCtx.status[this->objIndicies[2]].segment);
+}
```

Comment:
I feel like this function could have a better name. What about `_ChangeObject` or similar?